### PR TITLE
Resolve deprecation warning for pyOpenSSL PKCS#12

### LINKED ISSUE #360: [Bug]: PKCS#12 support in pyOpenSSL is deprecated
### Contact Details

_No response_

### Describe the Bug

During Unit Tests, the below warning for deprecation has appeared. Needs to be investigated and resolved.

edk2toolext/tests/test_pyopenssl_signer.py::Test_pyopenssl_signer::test_proper_options
  D:\a\edk2-pytool-extensions\edk2-pytool-extensions\edk2toolext\capsule\pyopenssl_signer.py:73: DeprecationWarning: PKCS#12 support in pyOpenSSL is deprecated. You should use the APIs in cryptography.
    pkcs12 = crypto.load_pkcs12(signer_options['key_data'])

### What Python version are you using?

Python 3.11

### Reproduction steps

1. Use Python 3.9, 3.10, 3.11
2. pip install --upgrade -r pip-requirements.txt
3. pip install -e .
4. `coverage run --include=edk2toolext\* -m pytest --junitxml=test.junit.xml --html=pytest_report.html --self-contained-html`
5. OR `pytest --junitxml=test.junit.xml --html=pytest_report.html --self-contained-html`

### Expected behavior

Deprecation Warning resolved

### Execution Environment

Windows, Linux

### Pip packages

pip-requirements.txt

### Additional context

Per CI action, this deprecation warning is confirmed in Python 3.9, 3.10, and 3.11

### PR DESCRIPTION
**Pull request by Javagedes**
1. Adds the ability to parse pfx files that have a password.
2. Resolves deprecation warning for pyOpenSSL PKCS#12 by switching .pfx reading from the pyOpenSSL library to the cryptography library.

Resolves #360

### CODE REVIEW

**Comment by spbrogan** [comment]
Overall, this looks good, but we might need to add a file to indicate that it is ok to check in the PFX file and that it is only a test key.   I guess Tianocore doesn't have that feature turned on.

**Comment by Javagedes** [comment]
@spbrogan If you're concerned with having a pfx file here, fake or not, I could serialize it and add it to the key_data entry instead of key_file in the test. Then I can have a comment right there that says this the key_data is fake data for testing purposes only.

File: edk2toolext/tests/test_pyopenssl_signer.py [hunk 1]
```diff
@@ -7,7 +7,223 @@
 ##
 import unittest
 from edk2toolext.capsule import pyopenssl_signer
-from OpenSSL import crypto
+
+# TEST_CERT is a cert used to test the ability to parse pfx and pull out necessary information.
+# To generate a new test cert follow these steps:
+# 1. Generate the Cert (.pfx) file'
+#   There are multiple ways to do this, but running this powershell script is the easiest:
```
Comment (spbrogan) [inline edk2toolext/tests/test_pyopenssl_signer.py:14]: can this be made a docstring using ''' so that it is easier to copy and paste

Comment (Javagedes) [reply edk2toolext/tests/test_pyopenssl_signer.py]: Updated to docstring

File: edk2toolext/tests/test_pyopenssl_signer.py [hunk 2]
```diff
@@ -6,8 +6,166 @@
...
+s5pMklXuIMGVe2PGLwVY2cbc4ms4m2PpyLBO0DqlZJ5G8hJuq11SE8jjfI3K\
+O9xc6D28mryoSTJE3ruN3mlRdbiAsNg15mUdXq9M9AthLaeSyCLT3a2QP1zt\
+O+AHsgd6ovMhfxAZ1R/0YayjtKB6Z4CN5rEAg2QdR5nvb6G4EKyqWAZhCCcd\
+UezFsMna+YaA/F2o9WIj9KPB+rdNh/KcbIJhpA2DSbbTPejHAk1TYBWall3B\
+2CfsFEIMjtuid/ggz/pUYCb88PN3BPoQ7GO/jv6Vi6F4oCQz+Y0srbmIStNj\
+qKXmc5OHIJoRkbrbFi4+2BeevncDADGrhl3heGWSmRlbOPxrryNrDPw7cbMS\
+UwIwYJKoZIhvcNAQkVMRYEFILT9pNixD3s66GdK3I48b/dr23DMDEwITAJBg\
+UrDgMCGgUABBTdiDC3a0y1gAbO1eZqveI3Zd0BrwQIFO8dSTjGLhMCAggA'
+
+# TEST_CERT2 has a password
```
Comment (spbrogan) [inline edk2toolext/tests/test_pyopenssl_signer.py:107]: Update TEST_CERT2 and TESTCERT2 to match.  Please do same for TEST_CERT1

**Review by spbrogan — APPROVED** [review]
Thanks.  Appreciate the work to get this into an easy-to-follow format.

File: edk2toolext/tests/test_pyopenssl_signer.py (hunk 2 above)
Comment (Javagedes) [reply edk2toolext/tests/test_pyopenssl_signer.py]: Updated

**Pull request merged**